src/ComplexType.php

```diff
@@ -117,7 +117,11 @@ protected function generateClass()
                 $getterCode = '  if ($this->' . $name . ' == null) {' . PHP_EOL
                     . '    return null;' . PHP_EOL
                     . '  } else {' . PHP_EOL
-                    . '    return \DateTime::createFromFormat(\DateTime::ATOM, $this->' . $name . ');' . PHP_EOL
+                    . '    try {' . PHP_EOL
+                    . '      return new \DateTime($this->' . $name . ');' . PHP_EOL
+                    . '    } catch (\Exception $e) {' . PHP_EOL
+                    . '      return null;' . PHP_EOL
```

Comment:
I would be good if you could replace `return null` with `return false`.

You are correct that returning `false` when expecting an object is not suitable. However `null` does not allow the caller to distinguish between an empty value and a failed conversion. For the next major version I think the correct way to handle this would be to throw an exception - probably by wrapping or not catching the exception from the DateTime constructor.